From 03c137273f334c9bd7c85d7a72d88d3e0bcaa417 Mon Sep 17 00:00:00 2001
From: "Jyri J. Virkki" <jyri@virkki.com>
Date: Tue, 2 Oct 2012 23:37:39 -0700
Subject: [PATCH] Add build support on Darwin.

---
 Makefile | 18 +++++++++++++-----
 test.c   |  5 +++--
 2 files changed, 16 insertions(+), 7 deletions(-)

diff --git a/Makefile b/Makefile
index e8f90b5..5b7f064 100644
--- a/Makefile
+++ b/Makefile
@@ -26,7 +26,7 @@ BUILD_OS := $(shell uname)
 BUILD=$(TOP)/build
 INC=-I$(TOP) -I$(TOP)/murmur2
 LIB=-lm
-CC=gcc ${OPT} ${MM} -std=c99 -fPIC
+CC=gcc -Wall ${OPT} ${MM} -std=c99 -fPIC
 
 ifeq ($(MM),)
 MM=-m32
@@ -34,10 +34,18 @@ endif
 
 ifeq ($(BUILD_OS),Linux)
 RPATH=-Wl,-rpath,$(BUILD)
+SO=so
 endif
 
 ifeq ($(BUILD_OS),SunOS)
 RPATH=-R$(BUILD)
+SO=so
+endif
+
+ifeq ($(BUILD_OS),Darwin)
+MAC=-install_name $(BUILD)/libbloom.dylib
+RPATH=-Xlinker -rpath -Xlinker $(BUILD)
+SO=dylib
 endif
 
 ifeq ($(DEBUG),1)
@@ -47,12 +55,12 @@ OPT=-O3
 endif
 
 
-all: $(BUILD)/libbloom.so $(BUILD)/test-libbloom
+all: $(BUILD)/libbloom.$(SO) $(BUILD)/test-libbloom
 
-$(BUILD)/libbloom.so: $(BUILD)/murmurhash2.o $(BUILD)/bloom.o
-	(cd $(BUILD) && $(CC) bloom.o murmurhash2.o -shared $(LIB) -o libbloom.so)
+$(BUILD)/libbloom.$(SO): $(BUILD)/murmurhash2.o $(BUILD)/bloom.o
+	(cd $(BUILD) && $(CC) bloom.o murmurhash2.o -shared $(LIB) $(MAC) -o libbloom.$(SO))
 
-$(BUILD)/test-libbloom: $(BUILD)/libbloom.so $(BUILD)/test.o
+$(BUILD)/test-libbloom: $(BUILD)/libbloom.$(SO) $(BUILD)/test.o
 	(cd $(BUILD) && $(CC) test.o -L$(BUILD) $(RPATH) -lbloom -o test-libbloom)
 
 $(BUILD)/%.o: %.c
diff --git a/test.c b/test.c
index 457925b..237de74 100644
--- a/test.c
+++ b/test.c
@@ -11,6 +11,7 @@
 #include <stdlib.h>
 #include <string.h>
 #include <sys/stat.h>
+#include <sys/time.h>
 #include <sys/types.h>
 #include <unistd.h>
 
@@ -115,8 +116,8 @@ static void perf_loop(int entries, int count)
   (void)gettimeofday(&tp, NULL);
   long after = (tp.tv_sec * 1000L) + (tp.tv_usec / 1000L);
 
-  (void)printf("Added %d elements of size %d, took %ldms (collisions=%d)\n",
-               count, sizeof(int), after - before, collisions);
+  (void)printf("Added %d elements of size %d, took %d ms (collisions=%d)\n",
+               count, (int)sizeof(int), (int)(after - before), collisions);
 
   (void)printf("%d,%d,%ld\n", entries, bloom.bytes, after - before);
   bloom_free(&bloom);
-- 
2.1.4

